src/block/icon-list/edit.js

```diff
@@ -243,12 +241,12 @@ const Edit = props => {
 
 		// Different icon sizes per device preview.
 		const { __experimentalGetPreviewDeviceType: getPreviewDeviceType } = select( 'core/edit-post' )
-		const previewDeviceType = getPreviewDeviceType()
+		const previewDeviceType = getPreviewDeviceType && getPreviewDeviceType()
```

Comment:
Instead of adding more checks, might be better to just assign it to `Desktop`
```suggestion
		const previewDeviceType = getPreviewDeviceType ? getPreviewDeviceType() : 'Desktop'
```